### PR TITLE
Add PyTorch🔥 documentation

### PR DESCRIPTION
**Pull request by n-poulsen**
# PyTorch Documentation

- This PR adds documentation to use the PyTorch engine.
- The PyTorch code is not quite released yet,  but this is being added to the docs for proper callouts
- Install docs are updated to remove [tf] callouts
- mac-specific conda file is not longer needed (WOO!) so it can be dropped (finally) // but leaving it, for now, for TF_only users
- adding course, expanding Beginner Guides
- restructured doc folders
- general clean up of outdated info.

### CODE REVIEW

**Comment by MMathisLab** [comment]
looks awesome @n-poulsen; I re-worked a bit the intro and in the architecture docs please check how I added BUCTD.... 
- need to do; add in superanimal models?

**Review by MMathisLab — APPROVED** [review]

**Review by AlexEMG — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]

File: docs/ModelZoo.md [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 ## 🏠 [Home page](http://modelzoo.deeplabcut.org/)
 
-Started in 2020 and expanded in 2022, the model zoo is four things:
+Started in 2020, expanded in 20222 with PhD student [Shaokai Ye et al.](https://arxiv.org/abs/2203.07436v1), and the first proper [SuperAnimal Foundation Models]() published in 2024 🔥, the Model Zoo is four things:
 
-- (1) a collection of models that are trained on diverse data across (typically) large datasets, which means you do not need to train models yourself
-- (2) a contribution website for community crowd sourcing of expertly labeled keypoints to improve models in part 1!
+- (1) a collection of models that are trained on diverse data across (typically) large datasets, which means you do not need to train models yourself, rather you can use them in your research applications.
+- (2) a contribution website for community crowd sourcing of expertly labeled keypoints to improve models! You can get involved here: [contrib.deeplabcut.org](https://contrib.deeplabcut.org/).
 - (3) a no-install DeepLabCut that you can use on ♾[Google Colab](https://colab.research.google.com/github/DeepLabCut/DeepLabCut/blob/master/examples/COLAB/COLAB_DLC_ModelZoo.ipynb),
```
Comment (yeshaokai) [inline docs/ModelZoo.md]: I think the colab here is pointing to the "old modelzoo", which has been causing confusion to the users. SuperAnimal modelzoo does not require create_pretrained_project and we have our own video inference API for it.

Comment (MMathisLab) [reply docs/ModelZoo.md]: already fixed.

**Review by MMathisLab — APPROVED** [review]

**Comment by MMathisLab** [comment]
@AlexEMG and @n-poulsen this can be merged now; the docs are clear on alpha PyTorch vs. stable TF (fow now), and have otherwise a lot of useful improvements.

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**